### ISSUE TITLE
DLangIDE fails to build

### ISSUE THREAD

**Issue by bad-dob** [dlang/dub#1515]
### System information
dub 1.7.0
Slackware linux 14.2
DMD64 D Compiler v2.078.1

### Bug Description
Fails to build

### How to reproduce?
$ git clone https://github.com/buggins/dlangide.git
$ cd dlangide
$ dub run

### Logs
The dependency resolution process is taking too long. The dependency graph is likely hitting a pathological case in the resolution algorithm. Please file a bug report at https://github.com/dlang/dub/issues and mention the package recipe that reproduces this error.

**Comment by WebFreak001** [comment]
#1513 should fix this

**Comment by WebFreak001** [comment]
confirmed getting error in old version and not getting it anymore now

**Comment by ZombineDev** [comment]
@kvetal there are 3 issues here:
1. "The dependency resolution process is taking too long..." - this was fixed with dub 1.11.0:
```
$ curl -fsS https://dlang.org/install.sh | bash -s dmd

$ source source ~/dlang/dmd-2.082.0/activate

$ dmd --version
DMD64 D Compiler v2.082.0
Copyright (C) 1999-2018 by The D Language Foundation, All Rights Reserved written by Walter Bright

$ dub --version
DUB version 1.11.0, built on Sep  1 2018

$ git clone https://github.com/buggins/dlangide.git

$ cd dlangide

$ git rev-parse HEAD
da20313a1b73d248807af744bc65bb6d277afb0c

$ dub run
Fetching dcd 0.9.1 (getting selected version)...
Fetching msgpack-d 1.0.0-beta.7 (getting selected version)...
Fetching dsfml 2.1.1 (getting selected version)...
Fetching libdparse 0.7.1 (getting selected version)...
Fetching dsymbol 0.2.9 (getting selected version)...
Fetching dlangui 0.9.178 (getting selected version)...
Fetching derelict-util 2.0.6 (getting selected version)...
Fetching x11 1.0.21 (getting selected version)...
Fetching derelict-gl3 1.0.24 (getting selected version)...
Fetching derelict-ft 1.1.3 (getting selected version)...
Fetching derelict-sdl2 2.0.2 (getting selected version)...
Fetching emsi_containers 0.5.3 (getting selected version)...
Performing "debug" build using /home/zlx/dlang/dmd-2.082.0/linux/bin64/dmd for x86_64.
emsi_containers 0.5.3: building configuration "library"...
libdparse 0.7.1: building configuration "library"...
[... warnings omitted for brevity]
dsymbol 0.2.9: building configuration "library"...
[... warnings omitted for brevity]
msgpack-d 1.0.0-beta.7: building configuration "library"...
dcd 0.9.1: building configuration "library"...
[... warnings omitted for brevity]
derelict-util 2.0.6: building configuration "library"...
derelict-ft 1.1.3: building configuration "library"...
derelict-gl3 1.0.24: building configuration "library"...
derelict-sdl2 2.0.2: building configuration "library"...
dlangui 0.9.178: building configuration "default"...
/home/zlx/.dub/packages/dlangui-0.9.178/dlangui/3rdparty/dimage/stream.d(226,16): Error: module `string` is in file 'std/c/string.d' which cannot be read
import path[0] = /home/zlx/.dub/packages/dlangui-0.9.178/dlangui/3rdparty
import path[1] = /home/zlx/.dub/packages/dlangui-0.9.178/dlangui/src
import path[2] = /home/zlx/.dub/packages/derelict-ft-1.1.3/derelict-ft/source/
import path[3] = /home/zlx/.dub/packages/derelict-util-2.0.6/derelict-util/source/
import path[4] = /home/zlx/.dub/packages/derelict-gl3-1.0.24/derelict-gl3/source/
import path[5] = /home/zlx/.dub/packages/derelict-sdl2-2.0.2/derelict-sdl2/source/
import path[6] = /home/zlx/dlang/dmd-2.082.0/linux/bin64/../../src/phobos
import path[7] = /home/zlx/dlang/dmd-2.082.0/linux/bin64/../../src/druntime/import
/home/zlx/dlang/dmd-2.082.0/linux/bin64/dmd failed with exit code 1.
```

2. `Error: module `string` is in file 'std/c/string.d'` - this was fixed with https://github.com/buggins/dlangui/pull/571 on Aug 8, but because the last tagged version (`v0.9.178`) is from May 8, which is why you're seeing the error above.

3. DlangIDE compilation warnings treated as errors:
```
src/dlangide/ui/newfile.d(100,101): Warning: C preprocessor directive #FF0000 is not supported
src/dlangide/ui/newfolder.d(64,101): Warning: C preprocessor directive #FF0000 is not supported
```

To fix the problems above, remove `dub.selections.json`,  force the dub to download the `~master` version of dlangui and allow compilation with warnings:
```
$ rm dub.selections.json

$ sed -i 's/==0.9.178/~master/g' dub.json

$ sed -i '14i\ \ \ \ "buildRequirements" : ["allowWarnings"],\n' dub.json

$ dub run
WARNING: A deprecated branch based version specification is used for the dependency dlangui. Please use numbered versions instead. Also note that you can still use the dub.selections.json file to override a certain dependency to use a branch instead.
Performing "debug" build using /home/zlx/dlang/dmd-2.082.0/linux/bin64/dmd for x86_64.
emsi_containers 0.5.3: target for configuration "library" is up to date.
libdparse 0.7.1: target for configuration "library" is up to date.
dsymbol 0.2.9: target for configuration "library" is up to date.
msgpack-d 1.0.0-beta.7: target for configuration "library" is up to date.
dcd 0.9.1: target for configuration "library" is up to date.
derelict-util 2.0.6: target for configuration "library" is up to date.
derelict-ft 1.1.3: target for configuration "library" is up to date.
derelict-gl3 1.0.24: target for configuration "library" is up to date.
derelict-sdl2 2.0.2: target for configuration "library" is up to date.
dlangui ~master: target for configuration "default" is up to date.
dlangide 0.8.12+commit.2.gda20313: building configuration "default"...
src/dlangide/ui/newfile.d(100,101): Warning: C preprocessor directive #FF0000 is not supported
src/dlangide/ui/newfolder.d(64,101): Warning: C preprocessor directive #FF0000 is not supported
[ ...deprecation messages omitted for brevity... ]
Linking...
To force a rebuild of up-to-date targets, run again with --force.
Running ./bin/dlangide
[dlangide output...]
```

So in summary this should do the trick for now:
```sh
# download and activate the latest dmd:
$ source $(curl https://dlang.org/install.sh | bash -s -- dmd -a)

$ git clone https://github.com/buggins/dlangide.git

$ cd dlangide

# use the master branch of dlangui:
$ sed -i 's/==0.9.178/~master/g' dub.json

# allow compilation warnings:
$ sed -i '14i\ \ \ \ "buildRequirements" : ["allowWarnings"],\n' dub.json

# fetch dependencies, build and run:
$ dub
```
